src/main/java/com/checkmarx/flow/controller/ADOController.java

```diff
@@ -302,10 +307,26 @@
         return emails;
     }
 
-    private String determineNamespace(Repository repository) {
-        String result = repository.getProject().getName().replace(" ", "_");
-        log.debug("Using namespace based on repository.project.name: {}", result);
-        return result;
+    private String determineNamespace(ResourceContainers resourceContainers) {
+        String namespace = EMPTY_STRING;
+        try {
+            log.debug("Trying to extract namespace from request body");
+            String projectUrl = resourceContainers.getProject().getBaseUrl();
+            namespace = projectUrl.split("/")[NAMESPACE_INDEX];
+        }
+        catch (Exception e){
+            log.warn("can't find namespace in body resource containers");
```

Comment:
I would use a specific exception instead of general one if possible + print the e.getMessage()